storage/storage_dest.go

```diff
@@ -313,13 +313,20 @@ func (f *zstdFetcher) GetBlobAt(chunks []chunked.ImageSourceChunk) (chan io.Read
 // It is available only if SupportsPutBlobPartial().
 // Even if SupportsPutBlobPartial() returns true, the call can fail, in which case the caller
 // should fall back to PutBlobWithOptions.
-func (s *storageImageDestination) PutBlobPartial(ctx context.Context, chunkAccessor private.BlobChunkAccessor, srcInfo types.BlobInfo, options private.PutBlobPartialOptions) (private.UploadedBlob, error) {
+func (s *storageImageDestination) PutBlobPartial(ctx context.Context, chunkAccessor private.BlobChunkAccessor, srcInfo types.BlobInfo, options private.PutBlobPartialOptions) (_ private.UploadedBlob, Err error) {
 	fetcher := zstdFetcher{
 		chunkAccessor: chunkAccessor,
 		ctx:           ctx,
 		blobInfo:      srcInfo,
 	}
 
+	var perr chunked.ErrFallbackToOrdinaryLayerDownload
+	defer func() {
+		if errors.As(Err, &perr) {
+			Err = private.NewErrFallbackToOrdinaryLayerDownload(perr.Unwrap())
```

Comment:
(Just noting the thought for the record: hypothetically the returned error might be a _wrapped_ Fallback… and this drops the error text added by the wrapping. That’s probably fine — and it doesn’t matter with current c/storage.)

Reply:
I had the same concern when I originally wrote it.  Since you've also pointed it out, I think it might be better to just wrap `retErr` as it won't hurt to have both wrapper types